tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,231 @@
+
+    # Get dataset information
+    if dataset:
+        # Get attributes from dataset object
+        attribute_specs = [attr.spec() for attr in dataset.attributes.stream()]
+        # Get dataset spec information
+        description = dataset.description
+        tags = dataset.tags
+        primary_keys = dataset.key_attribute_names
+    else:
+        attribute_specs = _create_specs(
+            attribute_names=attributes, attribute_types=attribute_types
+        )
+
+    dataset_exists = exists(client=client, dataset=dataset_name)
+    if not dataset_exists:
+        creation_spec = {
+            "name": dataset_name,
+            "description": description,
+            "keyAttributeNames": primary_keys,
+            "externalId": external_id,
+            "tags": tags,
+        }
+        client.datasets.create(creation_spec)
+    else:
+        raise ValueError(f"A dataset with name '{dataset_name}' already exists")
+
+    # Get new dataset
+    target_dataset = client.datasets.by_name(dataset_name)
+
+    # Get current dataset attributes
+    target_dataset_attributes = target_dataset.attributes
+
+    # Update attributes in dataset
+    for attribute in attribute_specs:
+        attr_spec_dict = attribute.to_dict()
+        attribute_name = attr_spec_dict["name"]
+        if attribute_name in primary_keys:
+            # This attribute already exists
+            continue
+        else:
+            # Create new attribute
+            target_dataset_attributes.create(attr_spec_dict)
+
+    return target_dataset
+
+
+def update(
+    *,
+    dataset: Dataset,
+    primary_keys: List[str],
+    attributes: Optional[List[str]] = None,
+    attribute_types: Optional[JsonDict] = None,
+    description: Optional[str] = None,
+    tags: Optional[List[str]] = None,
+) -> Dataset:
+    """Flexibly update a source dataset in Tamr. Will add/remove attributes to match input.
+       If no attrbute_types are passed in the default will be ARRAY STRING
+
+    Args:
+        dataset: An existing TUC dataset
+        primary_keys: one or more attributes for primary key(s) of new dataset
+        attributes: list of attribute names to add/keep for dataset
+        attribute_types: dict of attribute types, attribute name is key and type is value
+        description: updated text description of dataset, if None will not update
+        tags: updated tags for dataset, if None will not update tags
+
+    Returns:
+        Updated Dataset
+
+    Raises:
+        requests.HTTPError: If any HTTP error is encountered
+        ValueError: trying to alter a unified dataset
+        ValueError: Primary keys did not match Tamr dataset primary keys
+    """
+    dataset_name = dataset.name
+    if dataset.upstream_datasets():
+        raise ValueError(f"{dataset_name} is not a source dataset")
+
+    # Confirm primary keys match
+    if primary_keys != dataset.spec().to_dict()["keyAttributeNames"]:
```

Comment:
I didn't test this but something like this I think. Just trying to make things consistent?
```suggestion
    if primary_keys: 
        if primary_keys != dataset.spec().to_dict()["keyAttributeNames"]:
```

Reply:
Currently primary_keys are no longer an arg